packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryModule.kt

```diff
@@ -0,0 +1,467 @@
+    get() = mPermissions
+      ?.hasGrantedPermissions(WRITE_EXTERNAL_STORAGE)
+      ?.not() ?: false
+
+  private fun getManifestPermissions(writeOnly: Boolean): Array<String> {
+    return listOfNotNull(
+      WRITE_EXTERNAL_STORAGE,
+      READ_EXTERNAL_STORAGE.takeIf { !writeOnly },
+      ACCESS_MEDIA_LOCATION.takeIf { Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q }
+    ).toTypedArray()
+  }
+
+  private fun runActionWithPermissions(assetsId: List<String>, action: Action, promise: Promise) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+
+      val pathsWithoutPermissions = MediaLibraryUtils.getAssetsUris(mContext, assetsId)
+        .filter { uri ->
+          mContext.checkUriPermission(
+            uri,
+            Binder.getCallingPid(),
+            Binder.getCallingUid(), Intent.FLAG_GRANT_WRITE_URI_PERMISSION
+          ) != PackageManager.PERMISSION_GRANTED
+        }
+
+      if (pathsWithoutPermissions.isNotEmpty()) {
+        val deleteRequest =
+          MediaStore.createWriteRequest(mContext.contentResolver, pathsWithoutPermissions)
+        val activity = mActivityProvider.currentActivity
+
+        try {
+          mUIManager.registerActivityEventListener(this)
+          mAction = action
+          activity.startIntentSenderForResult(
+            deleteRequest.intentSender,
+            WRITE_REQUEST_CODE,
+            null,
+            0,
+            0,
+            0
+          )
+        } catch (e: SendIntentException) {
+          promise.reject(ERROR_UNABLE_TO_ASK_FOR_PERMISSIONS, ERROR_UNABLE_TO_ASK_FOR_PERMISSIONS_MESSAGE)
+          mAction = null
+        }
+        return
+      }
+    }
+    action.runWithPermissions(true)
+  }
+
+  private fun interface Action {
+    fun runWithPermissions(permissionsWereGranted: Boolean)
+  }
+
+  private inner class MediaStoreContentObserver(handler: Handler, private val mMediaType: Int) :
+    ContentObserver(handler) {
+
+    private var mAssetsTotalCount = getAssetsTotalCount(mMediaType)
+
+    override fun onChange(selfChange: Boolean) {
+      this.onChange(selfChange, null)
+    }
+
+    override fun onChange(selfChange: Boolean, uri: Uri?) {
+      val newTotalCount = getAssetsTotalCount(mMediaType)
+
+      // Send event to JS only when assets count has been changed - to filter out some unnecessary events.
+      // It's not perfect solution if someone adds and deletes the same number of assets in a short period of time, but I hope these events will not be batched.
+      if (mAssetsTotalCount != newTotalCount) {
+        mAssetsTotalCount = newTotalCount
+        mEventEmitter.emit(LIBRARY_DID_CHANGE_EVENT, Bundle())
+      }
+    }
+
+    private fun getAssetsTotalCount(mediaType: Int): Int {
+      return mContext.contentResolver.query(
+        EXTERNAL_CONTENT,
+        null,
+        "${MediaStore.Files.FileColumns.MEDIA_TYPE} == $mediaType",
+        null,
+        null
+      ).use { countCursor -> countCursor?.count ?: 0 }
+    }
```

Comment:
```suggestion
    private fun getAssetsTotalCount(mediaType: Int): Int =
      mContext.contentResolver.query(
        EXTERNAL_CONTENT,
        null,
        "${MediaStore.Files.FileColumns.MEDIA_TYPE} == $mediaType",
        null,
        null
      ).use { countCursor -> countCursor?.count ?: 0 }
```

[Kotlin coding conventions page](https://kotlinlang.org/docs/coding-conventions.html#functions) does not explicitly specify which style is better ==> Up to you!
(I believe this should be possible)